app/scripts/directives/overview/serviceInstanceRow.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+angular.module('openshiftConsole').component('serviceInstanceRow', {
+  controller: [
+    '$filter',
+    'DataService',
+    'rowMethods',
+    '$uibModal',
+    ServiceInstanceRow
+  ],
+  controllerAs: 'row',
+  bindings: {
+    apiObject: '<',
+    state: '<'
+  },
+  templateUrl: 'views/overview/_service-instance-row.html'
+});
+
+function ServiceInstanceRow($filter, DataService, rowMethods, $uibModal) {
+  var row = this;
+  _.extend(row, rowMethods.ui);
+
+  // returns the best user-friendly name available for the instance
+  var getDisplayName = function() {
+    var serviceClassName = row.apiObject.spec.serviceClassName;
+    var instanceName = row.apiObject.metadata.name;
+    var serviceClassDisplayName = _.get(row, ['state','serviceClasses', serviceClassName, 'OSBMetadata', 'displayName']);
+    return serviceClassDisplayName || serviceClassName || instanceName;
+  };
+
+  var getBindings = function() {
+    return _.filter(row.state.bindings, function(binding) {
+      return binding.spec.instanceRef.name === row.apiObject.metadata.name;
+    });
+  };
+
+  row.$onChanges = function() {
+    row.notifications = rowMethods.getNotifications(row.apiObject, row.state);
+    row.displayName = getDisplayName();
+    row.instanceBindings = getBindings();
+  };
+
+  row.getSecretForBinding = function(binding) {
+    return binding && row.state.secrets[binding.spec.secretName];
+  }
+
+
+  row.deprovision = function() {
+    // TODO: we probably have to handle bindings in here.
+    // either:
+    // - automatically remove the bindings
+    // - tell the user they must manually unbind before continue
+    $uibModal.open({
+      animation: true,
+      templateUrl: 'views/modals/confirm.html',
+      controller: 'ConfirmModalController',
+      resolve: {
+        modalConfig: function() {
+          return {
+            alerts: {
+              deprovision: {
+                type: 'error',
+                message: 'Service \'' + row.apiObject.spec.serviceClassName + '\' will be deleted and no longer available.'
+              }
+            },
+            detailsMarkup: 'Deprovision Service?',
+            okButtonText: 'Deprovision Service',
+            okButtonClass: 'btn-danger',
+            cancelButtonText: 'Cancel'
+          };
+        }
+      }
+    })
+    .result.then(function() {
+      DataService.delete({
+        group: 'servicecatalog.k8s.io',
+        resource: 'instances'
+      },
+      row.apiObject.metadata.name,
+      { namespace: row.apiObject.metadata.namespace })
+      .then(function(resp) {
+        console.log('instances delete success/deprovision', resp);
```

Comment:
clean up these two console.logs and handle the two cases correctly.  success should just be to close the dialog and add an inline alert to the page for now for consistency with our other actions.  failure should be showing a message in the dialog